### PR TITLE
Refactor support function transformations

### PR DESCRIPTION
**Pull request by knyazer**

### CODE REVIEW

**Comment by knyazer** [comment]
Now instead of passing the whole shapes into collision detection logic, only the support computations functions are passed.

Then, the transformation of the coordinate systems can be done naively with simply wrapping the support into the transformation, and passing the wrapped support as a real one.

The only thing left is to figure out what precisely to do with HomogenousTransformer. I would be happy if you could come up with something better than the current transformer approach, but I don't there is anything, without sacrificing performance

**Comment by knyazer** [comment]
@mikhail-vlasenko I would appreciate if you could take a look

**Comment by mikhail-vlasenko** [comment]
when would it be the case that the support function passed to collision detection is not `shape.get_support`?

**Comment by knyazer** [comment]
> when would it be the case that the support function passed to collision detection is not `shape.get_support`?

Exactly when using bodies collision detection methods: they use a support function, that is a local support function transformed into a different coordinate system

File: cotix/_bodies.py [hunk 1]
```diff
@@ -46,13 +46,19 @@ def get_transform(self):
...
         )
 
+    def get_global_support(self, direction):
+        """
+        Computes a support vector in a global coordinate system.
+        """
+        self._transform.inverse_direction(direction)
+        local_support = self.shape.get_support(direction)
+        return self._transform.forward_vector(local_support)
+
```
Comment (knyazer) [inline cotix/_bodies.py:56]: This function is the local support in the global CO

**Review by mikhail-vlasenko — APPROVED** [review]
lgtm

**Pull request merged**